### PR TITLE
Add authentication to nats

### PR DESCRIPTION
**Pull request by kumaranvpl**
# Description

Please include a summary of the change and specify which issue is being addressed. Additionally, provide relevant motivation and context.

Fixes # (issue number)

## Type of change

Please delete options that are not relevant.

- [ ] Documentation (typos, code examples, or any documentation updates)
- [ ] Bug fix (a non-breaking change that resolves an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a fix or feature that would disrupt existing functionality)
- [ ] This change requires a documentation update

## Checklist

- [ ] My code adheres to the style guidelines of this project (`scripts/lint.sh` shows no errors)
- [ ] I have conducted a self-review of my own code
- [ ] I have made the necessary changes to the documentation
- [ ] My changes do not generate any new warnings
- [ ] I have added tests to validate the effectiveness of my fix or the functionality of my new feature
- [ ] Both new and existing unit tests pass successfully on my local environment by running `scripts/test-cov.sh`
- [ ] I have ensured that static analysis tests are passing by running `scripts/static-anaylysis.sh`
- [ ] I have included code examples to illustrate the modifications

### CODE REVIEW

File: .devcontainer/python-3.11/docker-compose.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ services:
     environment:
       - DATABASE_URL=postgresql://admin:password@${USER}-postgres-py311-fastagency:5432/fastagency
       - PY_DATABASE_URL=postgresql://admin:password@${USER}-postgres-py311-fastagency:5432/pyfastagency
-      - NATS_URL=nats://${USER}-nats-py311-fastagency:4222
+      - NATS_URL=nats://faststream:faststream_password@${USER}-nats-py310-fastagency:4222
```
Comment (davorrunje) [inline .devcontainer/python-3.11/docker-compose.yml:13]: 310?

File: .devcontainer/python-3.10/docker-compose.yml [hunk 2]
```diff
@@ -22,6 +22,8 @@ services:
     # ports:
       # - "${PORT_PREFIX}4222:4222"
       # - "${PORT_PREFIX}9222:9222"
+    env_file:
```
Comment (davorrunje) [inline .devcontainer/python-3.10/docker-compose.yml:25]: why is this needed now?

File: .devcontainer/python-3.12/docker-compose.yml [hunk 3]
```diff
@@ -10,7 +10,7 @@ services:
     environment:
       - DATABASE_URL=postgresql://admin:password@${USER}-postgres-py312-fastagency:5432/fastagency
       - PY_DATABASE_URL=postgresql://admin:password@${USER}-postgres-py312-fastagency:5432/pyfastagency
-      - NATS_URL=nats://${USER}-nats-py312-fastagency:4222
+      - NATS_URL=nats://faststream:faststream_password@${USER}-nats-py310-fastagency:4222
```
Comment (davorrunje) [inline .devcontainer/python-3.12/docker-compose.yml:13]: 310?

File: .devcontainer/python-3.9/docker-compose.yml [hunk 4]
```diff
@@ -10,7 +10,7 @@ services:
     environment:
       - DATABASE_URL=postgresql://admin:password@${USER}-postgres-py39-fastagency:5432/fastagency
       - PY_DATABASE_URL=postgresql://admin:password@${USER}-postgres-py39-fastagency:5432/pyfastagency
-      - NATS_URL=nats://${USER}-nats-py39-fastagency:4222
+      - NATS_URL=nats://faststream:faststream_password@${USER}-nats-py310-fastagency:4222
```
Comment (davorrunje) [inline .devcontainer/python-3.9/docker-compose.yml:13]: 310?

**Review by davorrunje — CHANGES_REQUESTED** [review]

File: .devcontainer/python-3.10/docker-compose.yml (hunk 2 above)
Comment (kumaranvpl) [reply .devcontainer/python-3.10/docker-compose.yml:25]: In local also I used to run NATS without auth. 
Now I am setting up auth in local NATS using config file - https://github.com/airtai/fastagency/pull/278/files#diff-c13c0a1a0400d35b0786cde52030de4cdbaa32fb6df1e5a0584546b0ccf524af. 
In the config file I am using env variables to setup password for NATS users so I need to pass env file to NATS in order to set those env vars - https://github.com/airtai/fastagency/pull/278/files#diff-ddd7709d62c48ce521c3efe76e09432f4ce1aceefebb93b5353973c9dddbdb06

In staging and prod, I am not going to hardcode password instead going to use env variable. I have replicated the same approach in local devcontainer.
